src/tutorials/lua_basics_lightbulb.md

```diff
@@ -365,398 +390,227 @@ switchTrigger.interactedEvent:Connect(OnSwitchInteraction)
 
 ### Test the Variable Toggle
 
-Test to see if the script correctly toggles between `lightIsOn = false` and `lightIsOn = true` when the player interacts with the switch.
+Test to see if the script correctly toggles between `isSwitchOn = false` and `isSwitchOn = true` when the player interacts with the switch.
 
-Anywhere in our `OnSwitchInteraction` function type:
+In the `OnTriggerInteracted` function, after the `isSwitchOn = not isSwitchOn` statement, type:
 
 ```lua
-print(lightIsOn)
+print(isSwitchOn)
 ```
 
 - `print` tells the script to print to the Event Log window.
-- `lightIsOn` is the variable that will be printing.
+- `isSwitchOn` is the variable that will be printing.
 
 1. Open up the **Event Log** window from the **Window** menu in the top bar.
 2. Keep this open and press **Play**.
 3. Press ++F++ to interact with the switch.
 
+The OnTriggerInteracted function should look like:
+
+```lua
+local function OnTriggerInteracted()
+    isSwitchOn = not isSwitchOn
+    print(isSwitchOn)
+
+    Switch:RotateTo(ROTATION_ON, TIME_ROTATE, true)
+    Light.visibility = Visibility.INHERIT
+end
+```
+
 The **Event Log** should print `true` or `false` every time you interact with the light switch. You can delete the `print` statement now.
 
 ### Create an If Statement
 
 Now the script needs to know what to do specifically when the switch is on, and when it is not. An `if` statement will let you write code based on the state of other variables.
 
-In the `OnSwitchInteraction` function, after `lightIsOn = not lightIsOn` type:
+In the `OnTriggerInteracted` function, after the `lightIsOn = not lightIsOn` statement, type:
 
 ```lua
-    if not lightIsOn then
+    if isSwitchOn then
 
     end
 ```
 
 - `if` statements are handy when you need a certain series of actions to happen when a certain set of conditions is true.
-- `not lightIsOn` is the condition that must be met in order to execute the script inside our if statement.
+- `isSwitchOn` is the condition that must be met in order to execute the script inside our if statement.
 - `then` signifies the start of the code that will be performed if the conditions of the `if` statement are met.
 - `end` tells the script the `if` statement is over.
 
-Place the `switch:RotateTo()` statement and the `World.SpawnAsset()` function inside the `if` statement.
+Place the `Switch:RotateTo()` statement and the `Light.visibility` statement inside the `if` statement.
 
-Your `OnSwitchInteraction()` function should now look like this:
+Your `OnTriggerInteracted()` function should now look like this:
 
 ```lua
-local function OnSwitchInteraction()
-    if not lightIsOn then
-        switch:RotateTo(onRotation, 0.5, true)
-        World.SpawnAsset(propLightTemplate, {position = bulbPosition})
-    end
+function OnTriggerInteracted()
+    -- Flips the cached state of if the switch is on or off
```

Comment:
Take out cached. Saves a way for the script to look up of the switch is currently on or off